File: fixtures/scale/dart/large/lib/ciphers/beaufort/beaufort_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'beaufort_cipher.dart';
import 'beaufort_key.dart';
import 'beaufort_vectors.dart';

class BeaufortSuite implements CipherSuite {
  @override
  String name() => 'beaufort';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = BeaufortCipher(BeaufortKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in beaufortSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/beaufort/BeaufortSuite.kt
package cryptobench.ciphers.beaufort

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.roundTrip

class BeaufortSuite : CipherSuite {
    override fun name(): String = "beaufort"

    override fun run(): SuiteResult {
        val cipher: Cipher = BeaufortCipher(BeaufortKey.default())
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in BeaufortVectors.samples()) {
            if (roundTrip(cipher, sample)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}
